diff -u cfitsio/configure cfitsio-fixed/configure
--- cfitsio/configure	2007-11-05 15:52:57.000000000 -0500
+++ cfitsio-fixed/configure	2008-09-19 18:50:51.000000000 -0400
@@ -3397,7 +3397,8 @@
     SHLIB_SUFFIX=".sl"
     ;;
   lnx)
-    SHLIB_LD=":"
+    SHLIB_LD="cc -shared -Wl,-soname,libcfitsio.so.0"
+    SHLIB_SUFFIX=".so.0"
     ;;
   osf)
     SHLIB_LD="ld -shared -expect_unresolved '*'"
@@ -3418,7 +3419,7 @@
 # Darwin uses gcc (=cc), but needs different flags (see above)
 # if test "x$GCC" = xyes; then
 if test "x$GCC" = xyes && test "x$EXT" != xdarwin && test "x$EXT" != xcygwin; then
-  SHLIB_LD="$CC -shared"
+  SHLIB_LD="cc -shared -Wl,-soname,libcfitsio.so.0"
   lhea_shlib_cflags='-fPIC'
 fi
 if test "x$lhea_shlib_cflags" != x; then
diff -u cfitsio/Makefile.in cfitsio-fixed/Makefile.in
--- cfitsio/Makefile.in	2008-01-14 13:30:51.000000000 -0500
+++ cfitsio-fixed/Makefile.in	2008-09-19 18:51:41.000000000 -0400
@@ -22,8 +22,8 @@
 exec_prefix	= @exec_prefix@
 DESTDIR		= 
 CFITSIO_PREFIX	= $(prefix)
-CFITSIO_LIB	= $(DESTDIR)$(exec_prefix)/lib
-CFITSIO_INCLUDE	= $(DESTDIR)$(prefix)/include
+CFITSIO_LIB	= $(DESTDIR)$(exec_prefix)/$(LIBDIR)
+CFITSIO_INCLUDE	= $(DESTDIR)$(prefix)/$(INCLUDEDIR)
 INSTALL_DIRS	= $(DESTDIR)@INSTALL_ROOT@ ${CFITSIO_LIB} ${CFITSIO_LIB}/pkgconfig ${CFITSIO_INCLUDE}
 
 
